src/python/zquantum/optimizers/optimizers_test.py

```diff
@@ -14,15 +15,15 @@ class TestOptimizers(unittest.TestCase):
 
     def setUp(self):
         grid = ParameterGrid([[0.5, 1.5, 0.1], [0.5, 1.5, 0.1]])
-
+        self.cost_function = BasicCostFunction(rosen)
         self.scipy_optimizer = ScipyOptimizer(method="L-BFGS-B")
         self.grid_search_optimizer = GridSearchOptimizer(grid)
         self.cmaes_optimizer = CMAESOptimizer(options={"sigma_0": 0.1})
         self.optimizers = [self.scipy_optimizer, self.grid_search_optimizer, self.cmaes_optimizer]
 
     def test_optimization(self):
         for optimizer in self.optimizers:
-            results = optimizer.minimize(rosen, initial_params=[0, 0])
+            results = optimizer.minimize(self.cost_function, initial_params=[0, 0])
```

Comment:
I believe this should be `self.cost_function.evaluate` -> method for the cost function evaluation, not the `CostFunction` object

Reply:
I disagree.
There might be additional logic inside the `minimize` function for handling, that requires access to the whole `cost_function` object.
E.g. history of evaluations. Also, we might want to modify the cost function during the runtime, for example for the adaptive optimizers.

Reply:
If this is the case, then the scipy optimizer needs to be updated.

Reply:
Addressed this in: [`13cd5a`](https://github.com/zapatacomputing/z-quantum-optimizers/commit/13cd5a067549bf8f25ba3796ce73be98a31ebc4c)